src/CMakeLists.txt

```diff
@@ -9,9 +9,14 @@ add_library(gse
             half_float.cpp
             octet_string.cpp)
 
+set_target_properties(gse
+    PROPERTIES
+        CXX_STANDARD 17
+        CXX_STANDARD_REQUIRED YES
+        CXX_EXTENSIONS NO)
 target_compile_options(gse PRIVATE
-     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>: -Wpedantic -Wextra -Werror -Wall>
-     $<$<CXX_COMPILER_ID:MSVC>: /WX>)
+     $<$<OR:$<CXX_COMPILER_ID:Clang>,$<CXX_COMPILER_ID:AppleClang>,$<CXX_COMPILER_ID:GNU>>: -Wpedantic -Wextra -Wall>
```

Comment:
no `-Werror`?

Reply:
I prefer that, but the challenge is that some compiler somewhere might produce a warning in the future. Having that flag brings everything to a halt.

Maybe it would be good to enable it if gse is built standalone, but if fetched via FetchContent() this flag would cause problems for downstream projects if there's a warning.